Provide definitions that are missing on Solaris.

This patch should be sent upstream.

--- rtorrent-0.9.6/src/utils/directory.h.orig	2015-09-07 13:12:40.000000000 -0700
+++ rtorrent-0.9.6/src/utils/directory.h	2015-09-16 09:22:02.113830073 -0700
@@ -41,6 +41,32 @@
 #include <vector>
 #include <inttypes.h>
 
+#if defined(sun) || defined(__sun) || defined(__sun__)
+extern "C" {
+  enum
+  {
+    DT_UNKNOWN = 0,
+#define DT_UNKNOWN     DT_UNKNOWN
+    DT_FIFO = 1,
+#define DT_FIFO        DT_FIFO
+    DT_CHR = 2,
+#define DT_CHR         DT_CHR
+    DT_DIR = 4,
+#define DT_DIR         DT_DIR
+    DT_BLK = 6,
+#define DT_BLK         DT_BLK
+    DT_REG = 8,
+#define DT_REG         DT_REG
+    DT_LNK = 10,
+#define DT_LNK         DT_LNK
+    DT_SOCK = 12,
+#define DT_SOCK        DT_SOCK
+    DT_WHT = 14
+#define DT_WHT         DT_WHT
+  };
+};
+#endif
+
 namespace utils {
 
 struct directory_entry {
